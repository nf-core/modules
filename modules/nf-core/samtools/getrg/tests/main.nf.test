
nextflow_process {

    name "Test Process SAMTOOLS_GETRG"
    script "../main.nf"
    process "SAMTOOLS_GETRG"

    tag "modules"
    tag "modules_nfcore"
    tag "samtools"
    tag "samtools/getrg"

    test("test-samtools-getrg") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					file(process.out.readgroup[0][1]).readLines(),
					process.out.versions
					).match()
				}
            )
        }
    }

}
