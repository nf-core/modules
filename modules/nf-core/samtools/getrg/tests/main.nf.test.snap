{
    "test-samtools-getrg": {
        "content": [
            [
                "@RG\tID:1\tLB:lib1\tPL:ILLUMINA\tSM:test\tPU:barcode1"
            ],
            [
                "versions.yml:md5,40cf98b3daaf262b18e889ea734dbc40"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-16T08:11:06.35329516"
    }
}